kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/InMemoryTemplateInvocationContext.java

```diff
@@ -0,0 +1,104 @@
+                                        .withConfig("kms", IntegrationTestingKmsService.class.getSimpleName())
+                                        .withConfig("kmsConfig", Map.of("name", kmsId.toString()))
+                                        .withConfig("selector", TemplateKekSelector.class.getSimpleName())
+                                        .withConfig("selectorConfig", Map.of("template", template))
+                                        .build();
+                            }
+
+                            @Override
+                            public TestKekManager getTestKekManager() {
+                                return new TestKekManager() {
+                                    @Override
+                                    public CompletionStage<Void> generateKek(String alias) {
+                                        Objects.requireNonNull(alias);
+
+                                        return kms.resolveAlias(alias)
+                                                .handle((u, t) -> {
+                                                    if (t instanceof UnknownAliasException) {
+                                                        var kekId = kms.generateKey();
+                                                        kms.createAlias(kekId, alias);
+                                                        return null;
+                                                    }
+                                                    else {
+                                                        throw new AlreadyExistsException("key with alias " + alias + "already exists");
+                                                    }
+                                                });
+                                    }
+
+                                    @Override
+                                    public CompletionStage<Void> rotateKek(String alias) {
+                                        Objects.requireNonNull(alias);
+                                        return kms.resolveAlias(alias)
+                                                .thenApply(u -> {
+                                                    var kekId = kms.generateKey();
+                                                    kms.createAlias(kekId, alias);
+                                                    return null;
+                                                });
+
+                                    }
+                                };
+                            }
+                        };
+                    }
+                },
+                (BeforeTestExecutionCallback) extensionContext -> {
+                    kms = IntegrationTestingKmsService.newInstance().buildKms(new IntegrationTestingKmsService.Config(kmsId.toString()));
+                },
+                (AfterTestExecutionCallback) extensionContext -> {
+                    IntegrationTestingKmsService.delete(kmsId.toString());
+                });
+    }
```

Comment:
generally feels like a big telescoping set of braces to read, maybe worth extracting the TestContext resolver into a method or class to so it's easier to see the list of `Extensions`

Reply:
It got refactored several times, please take another look.